app/views/directives/key-value-editor.html

```diff
@@ -0,0 +1,238 @@
+                  group-by="groupByKind">
+                  <span ng-bind-html="source.metadata.name | highlight : $select.search"></span>
+                </ui-select-choices>
+              </ui-select>
+            </div>
+            <div class="ui-select">
+              <ui-select ng-model="entry.selectedValueFromKey" ng-required="true" on-select="valueFromKeySelected(entry, $select.selected)">
+                <ui-select-match placeholder="Select key">
+                  {{$select.selected}}
+                </ui-select-match>
+                <ui-select-choices repeat="key in entry.selectedValueFrom.data | keys">
+                  <span ng-bind-html="key | highlight : $select.search"></span>
+                </ui-select-choices>
+              </ui-select>
+            </div>
+          </div>
+        </div>
+
+        <!-- value help block -->
+        <span
+          class="help-block value-validation-error"
+          ng-show="(forms.keyValueEditor[uniqueForValue(unique, $index)].$error.pattern)">
+          <span class="validation-text">{{ entry.valueValidatorError || valueValidatorError}}</span>
+          <span ng-if="entry.valueValidatorErrorTooltip || valueValidatorErrorTooltip" class="help action-inline">
+            <a
+              aria-hidden="true"
+              data-toggle="tooltip"
+              data-placement="top"
+              data-original-title="{{entry.valueValidatorErrorTooltip || valueValidatorErrorTooltip}}"
+              title="{{entry.valueValidatorErrorTooltip || valueValidatorErrorTooltip}}">
+              <i class="{{entry.valueValidatorErrorTooltipIcon || valueValidatorErrorTooltipIcon}}"></i>
+            </a>
+          </span>
+        </span>
+        <span
+          class="help-block value-min-length"
+          ng-show="(forms.keyValueEditor[uniqueForValue(unique, $index)].$error.minlength)">
+          <span class="validation-text">Minimum character count is {{valueMinlength}}</span>
+        </span>
+      </div>
+
+      <div class="key-value-editor-buttons">
+        <span
+          ng-if="(!cannotSort) && (entries.length > 1)"
+          class="fa fa-bars sort-row"
+          role="button"
+          aria-label="Move row"
+          aria-grabbed="false"
+          as-sortable-item-handle></span>
+        <a
+          href=""
+          class="pficon pficon-close delete-row as-sortable-item-delete"
+          role="button"
+          aria-label="Delete row"
+          ng-hide="cannotDeleteAny || cannotDeleteSome(entry.name) || entry.cannotDelete"
+          ng-click="deleteEntry($index, 1)"></a>
+      </div>
+    </div>
+
+    <div
+      class="key-value-editor-entry form-group"
+      ng-if="(!cannotAdd)">
+      <a
+        href=""
+        class="add-row-link"
+        role="button"
+        aria-label="Add row"
+        ng-click="onAddRow()">{{ addRowLink }}</a>
+      <span
```

Comment:
Moar simpler:  a single span wrapping both the pipe and the link with ng-if="valueFromSelectorOptions.length